core/channel-upgrade.go

```diff
@@ -639,60 +706,74 @@ func queryCanTransitionToFlushComplete(ctx context.Context, chain interface {
 	}
 }
 
-func querySettledChannelUpgradePair(
-	srcCtx, dstCtx QueryContext,
-	src, dst interface {
+func querySettledChannelUpgrade(
+	queryCtx QueryContext,
+	logger_ *log.RelayLogger,
+	chain interface {
 		Chain
 		StateProver
 	},
 	prove bool,
-) (*chantypes.QueryUpgradeResponse, *chantypes.QueryUpgradeResponse, bool, error) {
-	logger := GetChannelPairLogger(src, dst)
-	logger = &log.RelayLogger{Logger: logger.With(
-		"src_height", srcCtx.Height().String(),
-		"dst_height", dstCtx.Height().String(),
+) (*chantypes.QueryUpgradeResponse, bool, error) {
+	logger := &log.RelayLogger{Logger: logger_.With(
```

Comment:
Better use `logger` as the function parameter name rather than `logger_`, because we don't want to accidentally access `logger_` after this point.